core/src/main/java/org/ehcache/Ehcache.java

```diff
@@ -83,9 +83,9 @@
 /**
  * @author Alex Snaps
  */
-public class Ehcache<K, V> implements Cache<K, V>, UserManagedCache<K, V>, PersistentUserManagedCache<K, V> {
+public class Ehcache<K, V> implements Cache<K, V>, UserManagedCache<K, V> {
 
-  private final StatusTransitioner statusTransitioner;
+  protected final StatusTransitioner statusTransitioner;
```

Comment:
Rather than making the variable protected, we can use checkMaintenance method and introduce other methods to do maintenance status transition. Just a thought !